### ISSUE TITLE
[BUG] API errors for forms not communicated to the user

### ISSUE THREAD

**Issue by Fedron** [ghostfolio/ghostfolio#3219]
**Bug Description**

When the client sends a request to the API and an error is returned such as 400 Bad Request when invalid form data is supplied there is no good indication of this. Currently the form just closes and that is it, you can only see the issue if you use Dev Tools and look at the console.

**To Reproduce**

1. Navigate to Portfolio > Activities
2. Create a new `BUY` activity
3. Set the quantity to a negative number

**Expected behavior**

The form should stay open and shouldn't close straight away. It would make sense the form would only close once it receives a successful status code from the API.

If the API returns a non-successful status code it would be useful to see what that was:
- A 500 error should probably just result in an error notification and closing(?) the dialog
- A 400 error (such as when providing a negative value for the quantity) would ideally keep the dialog open and highlight the field that caused the error and the message.

**Screenshots**

In a self-hosted environment all you see is this in the console:
![image](https://github.com/ghostfolio/ghostfolio/assets/40535546/5ccb5425-70aa-4e2d-ac9d-11162bc3e1cd)

**Environment**

- Ghostfolio Version 2.68.0
- Self-hosted
- Experimental Features disabled
- Browser: Edge 123.0.2420.65
- OS: Unraid

**Comment by dtslvr** [comment]
I don't like it if the dialog is closed when an error occurs. I would have to fill in everything again. How about invalidating the field for negative numbers (`fee`, `quantity`, `unitPrice`) so that the form cannot be submitted in the first place?

**Comment by Fedron** [comment]
I was thinking it would make sense to reuse the DTOs which already have validation logic. When a form is submitted, before the request is sent, the validation logic on the [DTO](https://github.com/ghostfolio/ghostfolio/blob/main/apps/api/src/app/order/update-order.dto.ts) is run and if there were errors the corresponding fields would be invalidated and no request would be sent to the API.

I could have a go at this for just the order form for now.

I'm not sure how many other forms also lack this validation communication but I think it would be useful to find that out and add this behaviour to the others?

**Comment by dtslvr** [comment]
The most important forms are in `CreateOrUpdateAccountDialog` and `CreateOrUpdateActivityDialog`.

Maybe you could try something like this? I haven't tested it though.

```
const userData = plainToClass(CreateUserDto, this.userForm.value);
const errors: ValidationError[] = await validate(userData);

if (errors.length > 0) {
  // There are validation errors, mark corresponding fields as invalid
  errors.forEach((error) => {
    const formControl = this.userForm.get(error.property);

    if (formControl) {
      formControl.setErrors({ serverError: error.constraints });
    }
  });
}
```

**Comment by Fedron** [comment]
I've created an initial PR #3230 that adds displays the validation error messages for the activity and account forms.

I did observe when testing some other changes in the admin control panel that the market data dialogue has the same issue in that it doesn't show validation errors. Although I would say arguably worse from a UX point of view? If you provide an invalid value (such as an invalid URL) and press "Save", then you get no feedback at all as in this case the dialog stays open so only see the error in Dev Tools.
